src/main/java/frc/robot/utils/ThreadedGyro.java

```diff
@@ -0,0 +1,59 @@
+package frc.robot.utils;
+
+import com.kauailabs.navx.frc.AHRS;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ThreadedGyro {
+    private final AHRS gyro;
+    private final AtomicBoolean shouldReset = new AtomicBoolean(false);
+    private volatile double lastGyro;
+    private volatile double angleAdjustment;
+    private boolean enabled = false;
+    private final ScheduledExecutorService executor;
+
+    public ThreadedGyro(AHRS gyro) {
+        this.gyro = gyro;
+        this.angleAdjustment = gyro.getAngleAdjustment();
+        this.executor = Executors.newScheduledThreadPool(1);
+    }
+
+    public void execute(){
+        executor.scheduleAtFixedRate(() -> {
+            gyro.setAngleAdjustment(getAngleAdjustment());
+            if (shouldReset.get()) {
+                gyro.reset();
+                shouldReset.set(false);
+            }
+            updateGyro();
+        },0,10, TimeUnit.MILLISECONDS);
+    }
+
+    private void updateGyro() {
+        lastGyro = (gyro.getAngle() % 360)  * -1;
+    }
+
+    public double getGyroValue(){
+        return lastGyro;
+    }
+
+    public void updateIfDisabled() {
+        if (!enabled){
+            updateGyro();
```

Comment:
This is also updating the gyro from another thread, which is not advisable unless AHRS is thread safe

Reply:
The gyro is _technicaly_ somewhat thread safe but I dont want to rely on it. I removed the method it was not needed